### ISSUE TITLE
Segfault while navigating fuse mounted cloud storage (also happens while navigating local folders)

### ISSUE THREAD

**Issue by dshanks01** [linuxmint/nemo#2005]
```
 * Nemo version (nemo --version)
Nemo 3.8.6
 * Is issue with desktop or windowed nemo?
Not sure what this question is asking. Nemo is running in a window on the Cinnamon 3.8.9 desktop
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
Mint 19.0
 * Graphics hardware *and* driver used
Hybrid graphics
Intel 2nd Generation Core Processor Family Integrated Graphics Controller, driver i915
Nvidia GF119M [NVS 4200M], Nvidia driver version 390.77-0ubuntu0.18.04.1
 * 32 or 64 bit
64 bit
 ```

**Issue**
I am using rclone 1.45 (https://rclone.org) to mount cloud storage (Microsoft OneDrive & Google Drive) in my home folder. While navigating folders within the mounted filesystems, nemo randomly crashes with a segmentation fault. In addition, navigation seems "slow" with nemo when navigating within the mounted filesystems. Neither problem happens when using caja, nautilus, or Thunar to navigate the filesystem. Crashes seem most frequent when moving up the directory tree, but may occur at any time. Crashes are usually preceded by several tens of seconds of non-responsiveness.

I do not know if this problem occurs only when using rclone, but the rclone process is unaffected by the crash (it doesn't respawn or anything).

**Steps to reproduce**
Use rclone to mount a Google Drive or Microsoft OneDrive cloud storage account to a mount point in your home directory.
Launch nemo
Navigate through the folders in the mounted filesystem using either or both of the folders in the Folders Pane or the folder tree traphic at the top between the menubar and the Favorites and Folders Panes.
You will probably sometimes experience long pauses when moving from folder to folder.
Eventually (within a dozen folder changes) nemo will crash with a segmentation fault.

**Expected behaviour**
Nemo should not crash
Navigation should be "snappier" (as it is in caja, nautilus, and thunar).

**Other information**
The problem is particularly prevalent when navigating into or out of a folder with a large number of files (100). The folders that cause the most trouble are full of photographs. I'm guessing that part of the problem is nemo trying to fetch each file (over the internet since it's cloud storage) to generate thumbnails.

**Comment by dshanks01** [comment]
Here's a stack trace after the segfault:
(gdb) info threads
  Id   Target Id         Frame 
* 1    Thread 0x7f5949ceb080 (LWP 4507) finish_info_provider (directory=0x55a832b5f9e0, file=0x0, 
    provider=0x55a831dbfc40) at ../../libnemo-private/nemo-directory-async.c:4212
  2    Thread 0x7f593b777700 (LWP 4508) 0x00007f5945a20bf9 in __GI___poll (fds=0x55a831c75c30, nfds=2, timeout=1335)
    at ../sysdeps/unix/sysv/linux/poll.c:29
  3    Thread 0x7f593913f700 (LWP 4512) syscall () at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
  4    Thread 0x7f593a568700 (LWP 4510) 0x00007f5945a20bf9 in __GI___poll (fds=0x55a831cb6de0, nfds=1, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
  5    Thread 0x7f5939d67700 (LWP 4690) 0x00007f59459ec0a8 in __getdents (fd=25, buf=0x7f5924037d30 "", nbytes=32768)
    at ../sysdeps/unix/sysv/linux/getdents.c:95
  6    Thread 0x7f593af76700 (LWP 4509) 0x00007f5945a20bf9 in __GI___poll (fds=0x7f59140014a0, nfds=4, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
(gdb) bt
#0  0x000055a830a8c113 in finish_info_provider (directory=0x55a832b5f9e0, file=0x0, provider=0x55a831dbfc40)
    at ../../libnemo-private/nemo-directory-async.c:4212
#1  0x000055a830a8c227 in info_provider_idle_callback (user_data=0x55a832b23d60)
    at ../../libnemo-private/nemo-directory-async.c:4251
#2  0x00007f59476691f5 in g_main_context_dispatch () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007f59476695c0 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007f594766964c in g_main_context_iteration () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007f5947c29e3d in g_application_run () at /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
#6  0x000055a8309e9ad2 in main (argc=1, argv=0x7ffe815ab0c8) at ../../src/nemo-main.c:104
(gdb)

**Comment by dshanks01** [comment]
I've now seen this problem while navigating my home folder. Stack trace shows segfault at the same place as above. Fuse filesystems are mounted, but I had not traversed to any of the fuse mounted filesystems at the time the segfault happened.

I can provide a crash report, but because I don't know if nemo caches any sensitive data (e.g. passwords) I don't want to attach it where anyone could download it.

**Comment by mtwebster** [comment]
I've spent quite a bit of time trying to track this down, I originally thought it was dropbox (extension)-specific, as for me it only occurred there.  The trace is the same though.

This gives me more to test against, thanks.

**Comment by mtwebster** [comment]
Can you list what extensions you have active?  Does this occur without any enabled?  If not, could you enable just one at time to see if a particular extension is the cause?

I have some theories what's happening.  I'm bringing up extensions because that particular section where the crash happens is related to them.  I think these requests for info, whatever they are, are taking so long to complete that they can outlive the view they were requested for.

Also, quite a bit was done regarding performance between 3.8 and 4.0

**Comment by dshanks01** [comment]
Extensions Installed:
ChangeColorFolder+NemoPython
Nemo Audio Tab
Nemo Emblems
Nemo Media Columns
Nemo Share
Nemo Seahorse
Nemo Fileroller
Nemo DropBox

When I disable them all the problem appears to either disappear, or become much less frequent. Performance is much snappier too. I was able to enter and leave even very large (> 1000 files) fuse mounted cloud folders multiple times without any problems. Previously, Nemo would segfault the first or second time I attempted to exit one of these large folders.

Enabling Nemo DropBox by itself did not cause the problems to re-appear.

Enabling Nemo Fileroller by itself did not cause the problems to re-appear.

Enabling Nemo Seahorse by itself did not cause the problems to re-appear.

Enabling Nemo Share by itself did not cause the problems to re-appear.

Enabling Nemo Media Columns by itself **did** cause the problems (both performance and segfault) to re-appear.

Enabling Nemo Emblems by itself did not cause the problems to re-appear.

Enabling Nemo Audio Tab by itself did not cause the problems to re-appear.

Enabling ChangeColorFolder+NemoPython by itself did not cause the problems to re-appear.

Looks like the problem is caused by enabling Nemo Media Columns and navigating into a cloud folder containing a large number of files. Contents of the files do not matter. I was able to reproduce the crash by creating a folder containing 100 empty files in my mounted OneDrive cloud folder and then jumping into and out of that folder with mouse clicks in Nemo. Using a folder of empty files, the snappiness problem goes away, but the crash always occurs within a dozen cycles into and out of the folder. The crash usually while moving out of the folder. 

I confirmed that it's the same crash location (libnemo-private/nemo-directory-async.c:4212).
